File: AppGeradorQRCodeJava/app/src/main/java/java/nivelamento/appgeradorqrcode/MainActivity.java
package java.nivelamento.appgeradorqrcode;

import androidx.appcompat.app.AppCompatActivity;

import android.graphics.Bitmap;
import android.graphics.Color;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;

import com.google.zxing.BarcodeFormat;
import com.google.zxing.WriterException;
import com.google.zxing.common.BitMatrix;
import com.google.zxing.qrcode.QRCodeWriter;
import com.google.zxing.qrcode.encoder.QRCode;

public class MainActivity extends AppCompatActivity {

    EditText editQRCode;
    Button btnGerarQRCode;
    ImageView imgQRCode;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initComponentes();

        btnGerarQRCode.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if(TextUtils.isEmpty(editQRCode.getText().toString())){
                    editQRCode.setError("Campo Obrigatório");
                    editQRCode.requestFocus();
                } else{
                    gerarQRCode(editQRCode.getText().toString());
                }
            }
        });

    }

    

    private void initComponentes() {

        editQRCode = findViewById(R.id.editQRCode);
        btnGerarQRCode = findViewById(R.id.btnGerarQRCode);
        imgQRCode = findViewById(R.id.imgQRCode);
    }
    
    private void gerarQRCode(String conteudoDoQRCode) {

        QRCodeWriter qrCode = new QRCodeWriter();
        
        try {

            BitMatrix bitMatrix = qrCode.encode(conteudoDoQRCode, BarcodeFormat.QR_CODE, 196, 196);
            
            int tamanho = bitMatrix.getWidth();
            int altura = bitMatrix.getHeight();

            Bitmap bitmap = Bitmap.createBitmap(tamanho, altura, Bitmap.Config.RGB_565);

            for (int x = 0; x < tamanho; x++) {

                for (int y = 0; y < altura; y++) {

                    bitmap.setPixel(x, y, bitMatrix.get(x, y) ? Color.BLACK : Color.WHITE);

                }
            }

            imgQRCode.setImageBitmap(bitmap);
                    
        }catch (WriterException e){
            e.printStackTrace();
        }
        
    }
    
}

File: AppGeradorQRCodeKotlin/app/src/main/java/java/nivelamento/appgeradorqrcodekotlin/MainActivity.kt
package java.nivelamento.appgeradorqrcodekotlin

import android.graphics.Bitmap
import android.graphics.Color
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.text.TextUtils
import android.widget.Button
import android.widget.EditText
import android.widget.ImageView
import com.google.zxing.BarcodeFormat
import com.google.zxing.WriterException
import com.google.zxing.qrcode.QRCodeWriter
import com.google.zxing.qrcode.encoder.QRCode

class MainActivity : AppCompatActivity() {

    var editQRCode: EditText? = null
    var btnGerarQRCode: Button? = null
    var imgQRCode: ImageView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initComponentes()

        btnGerarQRCode!!.setOnClickListener {

            if (TextUtils.isEmpty(editQRCode!!.text.toString())){

                //código - validar edittext

                editQRCode!!.error = "Campo Obrigatório"
                editQRCode!!.requestFocus()

            } else{

                //outro código - gerar qr code
                gerarQRCode(editQRCode!!.text.toString())

            }


        }

    }

    fun initComponentes() {

        editQRCode = findViewById(R.id.editQRCode)
        btnGerarQRCode = findViewById(R.id.btnGerarQRCode)
        imgQRCode = findViewById(R.id.imgQRCode)


    }

    fun gerarQRCode(conteudoQRCode: String) {

        val qrCode = QRCodeWriter()

        try {

            val bitMatrix = qrCode.encode(conteudoQRCode, BarcodeFormat.QR_CODE, 196, 196)
            val tamanho = bitMatrix.width
            val altura = bitMatrix.height
            
            val bitmap = Bitmap.createBitmap(tamanho, altura, Bitmap.Config.RGB_565)

            //enquanto o x não atingir o tamanho...
            for (x in 0 until tamanho){
                for (y in 0 until altura){
                    bitmap.setPixel(x, y, if(bitMatrix[x, y]) Color.BLACK else Color.WHITE)
                }

                imgQRCode!!.setImageBitmap(bitmap)
            }
            
        }catch (e: WriterException){

        }

    }
}
